packages/react/src/primitives/PasswordField/ShowPasswordButton.tsx

```diff
@@ -1,28 +1,28 @@
 import * as React from 'react';
 
-import { ShowPasswordButtonProps } from '../types';
 import { Button } from '../Button';
 import { ComponentClassNames } from '../shared/constants';
 import { IconVisibility, IconVisibilityOff } from '../Icon';
 import { SharedText } from '../shared/i18n';
+import { PrimitiveWithForwardRef, ShowPasswordButtonProps } from '../types';
 
 const ariaLabelText = SharedText.ShowPasswordButton.ariaLabel;
 
-export const ShowPasswordButton: React.FC<ShowPasswordButtonProps> = ({
-  fieldType,
-  size,
-  onClick,
-}) => {
+const ShowPasswordButtonPrimitive: PrimitiveWithForwardRef<
+  ShowPasswordButtonProps,
+  typeof Button
+> = ({ fieldType, size, ...rest }, ref) => {
   return (
     <Button
-      className={ComponentClassNames.FieldShowPassword}
-      onClick={onClick}
-      size={size}
+      {...rest}
       ariaLabel={
         fieldType === 'password'
           ? ariaLabelText.showPassword
           : ariaLabelText.hidePassword
       }
+      className={ComponentClassNames.FieldShowPassword}
+      ref={ref}
+      size={size}
```

Comment:
IMO `...rest` should be at the bottom of the JSX prop list to allow developers to override `className`, `size` and `ariaLabel`